server/events/approve_policies_command_runner.go

```diff
@@ -107,18 +109,51 @@ func (a *ApprovePoliciesCommandRunner) buildApprovePolicyCommandResults(ctx *com
 			}
 			teams = append(teams, userTeams...)
 		}
+		isAdmin := prjCmds[0].PolicySets.Owners.IsOwner(ctx.User.Username, teams)
+
+		for _, prjCmd := range prjCmds {
+			var prjErrs error
+			var prjPolicyStatus []models.PolicySetApproval
+			// Grab policy set status for project
+			for _, prjPullStatus := range ctx.PullStatus.Projects {
+				if prjCmd.Workspace == prjPullStatus.Workspace &&
+					prjCmd.RepoRelDir == prjPullStatus.RepoRelDir &&
+					prjCmd.ProjectName == prjPullStatus.ProjectName {
+					prjPolicyStatus = prjPullStatus.PolicyStatus
+				}
+			}
 
-		if !prjCmds[0].PolicySets.IsOwner(ctx.User.Username, teams) {
-			result.Error = fmt.Errorf("contact policy owners to approve failing policies")
-			return
-		}
-	}
-
-	var prjResults []command.ProjectResult
+			for _, policySet := range prjCmd.PolicySets.PolicySets {
+				isOwner := policySet.Owners.IsOwner(ctx.User.Username, teams) || isAdmin
+				for i, policyStatus := range prjPolicyStatus {
+					if policySet.Name == policyStatus.PolicySetName {
+						if policyStatus.Approvals == 0 {
+							continue
+						}
+						if isOwner {
+							prjPolicyStatus[i].Approvals = policyStatus.Approvals + 1
+						} else {
+							prjErrs = multierror.Append(fmt.Errorf("Policy set: %s user %s is not a policy owner. Please contact policy owners to approve failing policies", policySet.Name, ctx.User.Username))
+						}
+						if prjPolicyStatus[i].Approvals != 0 {
+							prjErrs = multierror.Append(prjErrs, fmt.Errorf("Policy set: %s requires %d approvals, have %d.", policySet.Name, policySet.ReviewCount, (0-prjPolicyStatus[i].Approvals)))
+						}
+					}
+				}
+			}
 
-	for _, prjCmd := range prjCmds {
-		prjResult := a.prjCmdRunner.ApprovePolicies(prjCmd)
-		prjResults = append(prjResults, prjResult)
+			prjResult := command.ProjectResult{
```

Comment:
Not sure that this is the right way to do this. See comment in project_command_runner.go for ApprovePolicies/doApprovePolicies

Reply:
Would there be any issue with attaching projectContext to commandContext?

Reply:
@nitrocode - Are you or someone else available for a chat on how to approach some aspects of this WRT standards?

Breaking policies into discreetly approvable sets seems to break some current structures/assumptions in the code so far.

Reply:
cc: @runatlantis/maintainers @krrrr38 (et al) if anyone has thoughts on this

Reply:
I've pushed a change which moves the logic over there. Hooked up/added context to a few places and added a VCS client to project_command_runner.